### ISSUE TITLE
Add a new rule to check for Extended CSS usage without the correct separator

### ISSUE THREAD

**Issue by Alex-302** [AdguardTeam/AGLint#193]
* `beverfood.com#$#a[href^="/bnlink/?bnid="] { remove: true; }`

`{ remove: true; }` can be used only with `#$?#`

* Warn, if used an extended CSS selector with `##` marker.
`##div:contains(a)`
`##div:has(> table:contains(a))`
Not including `:has()` without not native selectors, like `##div:has(> table)` / `#?#div:has(> table)`.

**Comment by scripthunter7** [comment]
This rule should also work vica-versa:
- if Extended CSS was used without the appropriate separator, like `##div:contains(foo)`
- and on the other hand, if plain CSS was used with Extended CSS separator: `#?#div`

In addition, the rule must also create a fix, which means replacing the separator

**Comment by AdamWr** [comment]
> * ```adblock
>   #?#div
>   ``` 
>   ↓↓↓ should be fixed as ↓↓↓
>   ```adblock
>   ##div
>   ```
> * ```adblock
>   #$?#div { padding-top: 0 !important; }
>   ```
>   ↓↓↓ should be fixed as ↓↓↓
>   ```adblock
>   #$#div { padding-top: 0 !important; }
>   ```

Regarding these two last examples.
I think that for apps in some cases it might still be valid.
The problem is that if there is element with inline style, for example like this:
```html
<h1 style="display: block !important; position: relative !important;">Example Domain</h1>
```
then in apps, the rule like this:
```adblock
##h1
```
will not hide this element, because inline style has higher specificity/priority.
So using:
```adblock
#?#h1
```
for this case might be a valid solution, the same for CSS (`#$?#`) rule.

I don't have anything against about warning that such rules can be used without `?`, but I think that it would be a good idea to take into consideration that these rules still might be valid in some cases.

**Comment by Alex-302** [comment]
@AdamWr Maybe better to change all extended css rules **with native** `:has()` once? The main purpose of this - using native `:has()`.

**Comment by AdamWr** [comment]
I think that changing rules to use native `:has()` is a good idea.

I'm not sure if it was related to my comment above, but if so, then I meant rule like this:
```adblock
#?#h1
```
there is no `:has()`, and it might be used intentionally in the case of apps, to hide element with inline style like in example above, or in the case where for some other reason applying CSS rules do not work or stylesheet has been removed.

**Comment by Alex-302** [comment]
> there is no `:has()`, and it might be used intentionally in the case of apps, to hide element with inline style like in example above

Rules like this we will not change.
